spec/plaindate.html

```diff
@@ -374,6 +374,8 @@ <h1>Temporal.PlainDate.prototype.with ( _temporalDateLike_ [ , _options_ ] )</h1
         1. Perform ? RequireInternalSlot(_temporalDate_, [[InitializedTemporalDate]]).
         1. If Type(_temporalDateLike_) is not Object, then
           1. Throw a *TypeError* exception.
+        1. If _temporalDateLike_ has an [[InitializedTemporalDate]], [[InitializedTemporalDateTime]], [[InitializedTemporalMonthDay]], [[InitializedTemporalTime]], [[InitializedTemporalYearMonth]], or [[InitializedTemporalZonedDateTime]] internal slot, then
+          1. Throw a *TypeError* exception.
```

Comment:
these three operations seem kind of like it would be valuable as some kind of `Perform ? RequireTemporalType(_temporalDate_, ~PlainDate~)` rather than laboriously listing internal slots everywhere this kind of check is needed.

Reply:
I've gone with RejectTemporalCalendarType, hopefully the name makes it clear that the intention is that anything with a calendar should not make it through.